framework_agreement_sourcing/model/logistic_requisition.py

```diff
@@ -130,127 +116,57 @@ def _generate_lines_from_agreements(self, cr, uid, container, line,
-        line
-
-        :param line: requisition line browse record
-        :returns: (generated line ids, remaining qty not covered by agreement)
-
-        """
-        Sourced = namedtuple('Sourced', ['generated', 'remaining'])
-        qty = line.requested_qty
-        generated = []
-        remaining_qty = self._generate_lines_from_agreements(cr, uid,
-                                                             generated, line,
-                                                             agreements, qty,
-                                                             currency=currency,
-                                                             context=context)
-        return Sourced(generated, remaining_qty)
-
-    def make_source_line(self, cr, uid, line, force_qty=None, agreement=None,
-                         context=None):
-        """Generate a source line from a requisition line, see
-        _prepare_line_source for details.
-
-        :param line: browse record of origin logistic.request
-        :param force_qty: if set this quantity will be used instead
-        of requested quantity
-        :returns: id of generated source line
-
-        """
-        qty = force_qty if force_qty else line.requested_qty
-        src_obj = self.pool['logistic.requisition.source']
-        vals = self._prepare_line_source(cr, uid, line,
-                                         qty=qty,
-                                         agreement=agreement,
-                                         context=None)
-        return src_obj.create(cr, uid, vals, context=context)
-
-    def _generate_source_line(self, cr, uid, line, context=None):
+        agreements = self._sort_agreements(agreements, qty,
+                                           currency=currency)
+        src_model = self.env['logistic.requisition.source']
+        remaining_qty = qty
+        while remaining_qty and agreements:
+            current_agr = agreements.pop(0)
+            avail = current_agr.available_quantity
+            if not avail:
+                continue
+            avail_sold = avail - remaining_qty
+            to_consume = remaining_qty if avail_sold >= 0 else avail
+            remaining_qty -= to_consume
+
+            values = self._prepare_agreement_source(current_agr,
+                                                    qty=to_consume)
+            src_model.create(values)
+
+        return remaining_qty
+
+    @api.multi
+    def _generate_sources(self):
         """Generate one or n source line(s) per requisition line.
 
         Depending on the available resources. If there is framework
         agreement(s) running we generate one or n source line using agreements
         otherwise we generate one source line using tender process
 
-        :param line: browse record of origin logistic.request
-
-        :returns: list of generated source line ids
+        :param self: origin requisition line
```

Comment:
Nothing is returned anymore this because sources can be get from logistics requisition line source_ids field.